### PR TITLE
[fix] Fix test expectations for translator to avoid depending on order

### PR DESCRIPTION
**Pull request by mahadzaryab1**
## Description of the changes
- The output of `ProtoToTraces` is not guaranteed to be in order which was causing flaky test failures in `translator_test.go`. This PR changes the test expectations to not depend on the return order of `ProtoToTraces`. 

## How was this change tested?
- CI

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `npm run lint` and `npm run test`

### CODE REVIEW

**Comment by yurishkuro** [comment]
> The output of ProtoToTraces is not guaranteed to be in order

can you elaborate what this means?

**Comment by mahadzaryab1** [comment]
> > The output of ProtoToTraces is not guaranteed to be in order
> 
> can you elaborate what this means?

If we pass in `batch1,batch2` - there's no guarantee that resources.At(0) == batch1. They can be in either order. I think its because [this](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/translator/jaeger/jaegerproto_to_traces.go#L61) function maintains a map which it then iterates over to create the new set of batches to iterate over.

File: internal/jptrace/translator_test.go [hunk 1]
```diff
@@ -88,28 +88,28 @@ func TestProtoToTraces_AddsWarnings(t *testing.T) {
 
 	assert.Equal(t, 2, traces.ResourceSpans().Len())
 
-	rs1 := traces.ResourceSpans().At(0)
-	assert.Equal(t, 1, rs1.ScopeSpans().Len())
-	ss1 := rs1.ScopeSpans().At(0)
-	assert.Equal(t, 2, ss1.Spans().Len())
+	spanMap := make(map[string]ptrace.Span)
+
+	for i := 0; i < traces.ResourceSpans().Len(); i++ {
```
Comment (yurishkuro) [inline internal/jptrace/translator_test.go:93]: considering how often we seem to be doing this I would create a helper 
```go
package jptrace

func SpanMap[K](tr ptrace.Traces, keyFn func (ptrace.Span) K) map[K]ptrace.Span
```

**Review by yurishkuro — APPROVED** [review]

File: internal/jptrace/translator_test.go (hunk 1 above)
Comment (mahadzaryab1) [reply internal/jptrace/translator_test.go:93]: sounds good - can do in a follow-up PR

**Pull request merged**